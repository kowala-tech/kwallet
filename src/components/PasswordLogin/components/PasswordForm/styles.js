const styles = (theme) => ({
	buttonWrapper: {
		position: "relative",
		marginTop: theme.spacing.unit * 2
	},
	buttonProgress: {
		position: "absolute",
		top: "50%",
		left: "50%",
		marginTop: -12,
		marginLeft: -12,
	}
});

export default styles;
