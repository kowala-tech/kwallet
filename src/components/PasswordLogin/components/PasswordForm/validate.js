const validate = values => {
	const errors = {};
	// Username
	if (!values.username) {
		errors.username = "Username is required.";
	}
	// Password
	if (!values.password) {
		errors.password = "Password is required.";
	}
	return errors;
};

export default validate;
