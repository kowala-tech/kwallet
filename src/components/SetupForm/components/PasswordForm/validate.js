const validate = values => {
	const errors = {};
	// Password
	if (!values.password) {
		errors.password = "Password is required.";
	} else if (values.password.length < 8) {
		errors.password = "Password must be at least 8 characters.";
	}
	// Password Confirmation
	if (values.password !== values.passwordConfirmation) {
		errors.passwordConfirmation = "Passwords do not match.";
	}
	return errors;
};

export default validate;
