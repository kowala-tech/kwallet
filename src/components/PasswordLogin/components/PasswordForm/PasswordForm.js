import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
import classNames from "classnames";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";
// Form Imports
import { Field, reduxForm } from "redux-form";
import ReduxFormField from "../../../ReduxFormField";
import validate from "./validate";
// Component Related Imports
import styles from "./styles";

class PasswordForm extends React.Component {
	render() {
		const {
			classes,
			handleSubmit,
			invalid,
			pristine,
			touched,
			error,
			loading
		} = this.props;

		return (
			<form onSubmit={handleSubmit}>
				<Field
					name="username"
					type="username"
					component={ReduxFormField}
					label="Username"
					disabled={loading}
					fullWidth
					className={classNames({
						animated: (touched && error),
						headShake: (touched && error),
					})}
				/>
				<Field
					name="password"
					type="password"
					component={ReduxFormField}
					label="Password"
					disabled={loading}
					fullWidth
					className={classNames({
						animated: (touched && error),
						headShake: (touched && error),
					})}
				/>
				{error && <strong>{error}</strong>}
				<div className={classes.buttonWrapper}>
					<Button
						fullWidth
						variant="raised"
						color="primary"
						type="submit"
						disabled={loading || invalid || pristine}
					>
						{"Login"}
					</Button>
					{loading && <CircularProgress size={24}
						className={classes.buttonProgress} />}
				</div>
			</form>
		);
	}
}

PasswordForm.propTypes = {
	classes: PropTypes.object.isRequired,
	onSubmit: PropTypes.func.isRequired,
	input: PropTypes.object,
	label: PropTypes.object,
	meta: PropTypes.object,
	handleSubmit: PropTypes.func.isRequired,
	invalid: PropTypes.bool,
	pristine: PropTypes.bool,
	error: PropTypes.string,
	touched: PropTypes.bool,
	loading: PropTypes.bool
};

export default compose(
	withStyles(styles),
	reduxForm({
		form: "passwordForm",
		validate
	})
)(PasswordForm);
