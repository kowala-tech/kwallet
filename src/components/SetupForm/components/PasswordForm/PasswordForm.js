import React from "react";
import { compose } from "recompose";
import PropTypes from "prop-types";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
// Form Imports
import { Field, reduxForm } from "redux-form";
import validate from "./validate";
import ReduxFormField from "../../../ReduxFormField";
// Component Related Imports
import styles from "../../styles";

class PasswordForm extends React.Component {
	render() {
		const {
			classes,
			handleSubmit,
			invalid,
			pristine,
			previousStep
		} = this.props;
		return (
			<form onSubmit={handleSubmit}>
				<Field
					name="password"
					type="password"
					component={ReduxFormField}
					label="Password"
				/>
				<Field
					name="passwordConfirmation"
					type="password"
					component={ReduxFormField}
					label="Password Confirmation"
				/>
				<div className={classes.actionsContainer}>
					<Button
						onClick={previousStep}
						className={classes.button}
					>
						Back
					</Button>
					<Button
						variant="raised"
						color="primary"
						className={classes.button}
						type="submit"
						disabled={invalid || pristine}
					>
						Next
					</Button>
				</div>
			</form>
		);
	}
}

PasswordForm.propTypes = {
	classes: PropTypes.object.isRequired,
	onSubmit: PropTypes.func.isRequired,
	previousStep: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	invalid: PropTypes.bool.isRequired,
	pristine: PropTypes.bool.isRequired
};

export default compose(
	withStyles(styles),
	reduxForm({
		form: "setupForm",
		destroyOnUnmount: false,
		forceUnregisterOnUnmount: true,
		validate
	})
)(PasswordForm);
