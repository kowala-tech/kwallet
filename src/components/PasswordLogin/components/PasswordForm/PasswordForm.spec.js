import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import PasswordForm from "./PasswordForm";

describe("PasswordForm", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<PasswordForm
					onSubmit={jest.fn()}
					handleSubmit={jest.fn()}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
