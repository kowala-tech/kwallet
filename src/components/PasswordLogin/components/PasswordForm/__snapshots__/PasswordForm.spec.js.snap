// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForm it should match the snapshot 1`] = `
<ReduxForm
  classes={
    Object {
      "buttonProgress": "ReduxForm-buttonProgress-2",
      "buttonWrapper": "ReduxForm-buttonWrapper-1",
    }
  }
  handleSubmit={[MockFunction]}
  onSubmit={[MockFunction]}
/>
`;
